server/index.js

```diff
@@ -1,7 +1,7 @@
 const express = require("express");
 require("dotenv").config();
 const app = express();
-const port = 3000;
+const port = 4444; // 4444 for local dev, 3000 for Docker
```

Comment:
How does this work?
Do I have to change it manually here to 3000 if I use docker, and back to 4444 if I do not?

Reply:
This file is used for local development when `npm run server` is run. There's another server file in [docker/server/index.js](https://github.com/vectara/vectara-answer/blob/main/docker/server/index.js) which is used by Docker. I just added this note here in case someone (probably me!) thinks they can make a change here and just copy it over to the Docker server file including this line. That would break the Docker server. The files are identical except for this one line, so it will be an easy mistake to make.